packages/samples/headless-react/src/components/result-list/result-list.class.tsx

```diff
@@ -0,0 +1,58 @@
+import {Component} from 'react';
+import {
+  buildResultList,
+  ResultList as ResultListController,
+  ResultListState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {ResultLink} from './result-link';
+import {engine} from '../../engine';
+
+export class ResultList extends Component {
+  private readonly controller: ResultListController;
+  public state: ResultListState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: {}) {
+    super(props);
+
+    this.controller = buildResultList(engine);
+    this.state = this.controller.state;
+  }
+
+  componentDidMount() {
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  render() {
+    if (!this.state.results.length) {
+      return <div>No results</div>;
+    }
+
+    return (
+      <div>
+        <ul style={{textAlign: 'left'}}>
+          {this.state.results.map((result) => (
+            <li key={result.uniqueId} onLoad={(e) => console.log(e.target)}>
+              <article>
+                <h2>
+                  {/* It's important not to use a bare-bones anchor element in order to log analytics */}
```

Comment:
I'm not sure whether this comment is necessary or not, and what it should be saying.

Reply:
Hum 🤔 
Yeah, it is definitely interesting/import IMO.

Maybe just:
```
Pay close attention to the way click events are properly logged to the Coveo platform.
```

@kaitlynkatz  would you have a suggestion ?

Reply:
If I understand correctly, you're trying to warn that the component displaying a link in the results list needs to be properly configured in order to log analytics? Instead of just using a plain `<a>` element, for example.

Suggestion: `Make sure to log analytics when the result link is clicked.`